### PR TITLE
Fix SerEnum member values in generated P4Info

### PR DESCRIPTION
**Pull request by antoninbas**
The value should be a P4Runtime "bytestring" and not a human-readable
hexadecimal string.

Thanks to @mbudiu-vmw for his help making the SerEnum Google Test work.

### CODE REVIEW

File: test/gtest/p4runtime.cpp [hunk 1]
```diff
@@ -1660,6 +1660,36 @@ TEST_F(P4RuntimeDataTypeSpec, Enum) {
     EXPECT_EQ("MBR_2", it->second.members(1).name());
 }
 
+// Currently throws an exception:
```
Comment (antoninbas) [inline test/gtest/p4runtime.cpp]: @mbudiu-vmw I could use your help with this test.

I thought I would add a test for serializable enums, just like I wrote a while ago for enums, but the test throws a Compiler Bug in the type checker for some reason. If I write a similar program and run it through `p4test`, there is no issue. The different is that these tests only run the `TypeChecking` pass (see line 1409) and not the whole frontend. Is there another pass we should run? What's so special about serializable enums here?

File: testdata/p4_16_samples_outputs/v1model-digest-containing-ser-enum.p4.p4info.txt [hunk 2]
```diff
@@ -263,15 +263,15 @@ type_info {
...
       }
       members {
         name: "bar"
-        value: "17"
+        value: "\021"
       }
       members {
         name: "gah"
-        value: "42"
+        value: "*"
```
Comment (jafingerhut) [inline testdata/p4_16_samples_outputs/v1model-digest-containing-ser-enum.p4.p4info.txt]: Where is this "*" coming from?  Judging by the changes several lines above, shouldn't it be `"\052"`, i.e. escaped octal for the decimal value 42?

Comment (jafingerhut) [reply testdata/p4_16_samples_outputs/v1model-digest-containing-ser-enum.p4.p4info.txt]: Oh, I guess judging from the previous file's diffs, `"*"` has the ASCII value of decimal 42?  Kind of confusing the first time you see this.

Comment (antoninbas) [reply testdata/p4_16_samples_outputs/v1model-digest-containing-ser-enum.p4.p4info.txt]: yes these values prioritize size over human-readability :)

I got confused when I reviewed my diff as well...

**Comment by mbudiu-vmw** [comment]
how can cpplint complain for unified builds but not for non-unified builds?

**Comment by mbudiu-vmw** [comment]
@antoninbas : here is a fix that makes it work.
Normally the typechecker pass behaves like a Visitor, but you need to make it behave like a Transform, because it needs to change some nodes in the IR. That's the 'false' third argument.

**Comment by mbudiu-vmw** [comment]
https://github.com/p4lang/p4c/commit/ac366556e288fc2d29d9485100050b1a74e49d06

**Comment by antoninbas** [comment]
Thanks a lot @mbudiu-vmw, I appreciate you taking the time to look into this.

I'm guessing this line is mutating the IR so SerEnum members: https://github.com/p4lang/p4c/blob/master/frontends/p4/typeChecking/typeChecker.cpp#L1243

**Comment by mbudiu-vmw** [comment]
Yes, inferring a type for int constants is one of the most frequent reasons for the typechecker to mutate the program.

**Comment by antoninbas** [comment]
@mbudiu-vmw regarding the cpplint error, it only shows up for the unified builds because cpplint is not run for non-unified builds.

**Pull request merged**